### PR TITLE
python3Packages.typing-extensions: 4.5.0 -> 4.7.1

### PR DESCRIPTION
**Pull request by blaggacao**
Moved to target staging due to mass-rebuild: https://github.com/NixOS/nixpkgs/pull/243861

###### Description of changes

https://github.com/python/typing_extensions/blob/main/CHANGELOG.md

This will trigger a mass-rebuild. Help seeing this through is greatly appeciated.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/typing-extensions/default.nix [hunk 1]
```diff
@@ -8,15 +8,15 @@
 
 buildPythonPackage rec {
   pname = "typing-extensions";
-  version = "4.5.0";
+  version = "4.7.1";
   format = "pyproject";
 
   disabled = pythonOlder "3.6";
```
Comment (fabaff) [inline pkgs/development/python-modules/typing-extensions/default.nix:14]:
```suggestion
  disabled = pythonOlder "3.7";
```

see https://github.com/python/typing_extensions/blob/4.7.1/pyproject.toml#L12

**Review by fabaff — COMMENTED** [review]
Please update the `description` (best would be to use something like "Backported and Experimental Type Hints for Python") and add `meta.changelog`.

File: pkgs/development/python-modules/typing-extensions/default.nix (hunk 1 above)
Comment (blaggacao) [reply pkgs/development/python-modules/typing-extensions/default.nix]: Done in [force-push](https://github.com/NixOS/nixpkgs/compare/7fc3327aecfd10adc917b265e23e3d0b12d2cf90..ebccac00dbf2301b08cc5aa67504e65743b24325).

Maybe worth a headsup that `4.7.*` will also be the last version to support `3.7` moving on to `3.8`-only afterwards.

**Comment by blaggacao** [comment]
Moved to target staging due to mass-rebuild: https://github.com/NixOS/nixpkgs/pull/243861

**Pull request closed without merging**